toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1561,32 +1615,43 @@ func ProvisionUserSSHCerts(installChroot *safechroot.Chroot, username string, ss
 	}
 
 	// Change ownership of the folder to belong to the user and their primary group
-	err = installChroot.UnsafeRun(func() (err error) {
-		// Find the primary group of the user
-		stdout, stderr, err := shell.Execute("id", "-g", username)
-		if err != nil {
-			logger.Log.Warnf(stderr)
-			return
-		}
-
-		primaryGroup := strings.TrimSpace(stdout)
-		logger.Log.Debugf("Primary group for user (%s) is (%s)", username, primaryGroup)
+	err = installChroot.UnsafeRun(func() error {
+		err := setSSHDirOwnershipAndPermissions(username, userSSHKeyDir)
 
-		ownership := fmt.Sprintf("%s:%s", username, primaryGroup)
-		err = shell.ExecuteLive(squashErrors, "chown", "-R", ownership, userSSHKeyDir)
 		if err != nil {
-			return
+			return err
 		}
 
-		err = shell.ExecuteLive(squashErrors, "chmod", "-R", sshDirectoryPermission, userSSHKeyDir)
-		return
+		return err
 	})
 
+	return nil
+}
+
+func setSSHDirOwnershipAndPermissions(username, userSSHKeyDir string) error {
+	const squashErrors = false
+	const sshDirectoryPermission = "0700"
+
+	// Find the primary group of the user
+	stdout, stderr, err := shell.Execute("id", "-g", username)
 	if err != nil {
-		return
+		return fmt.Errorf("error getting primary group for %s: %s", username, stderr)
 	}
+	primaryGroup := strings.TrimSpace(stdout)
+	logger.Log.Debugf("Primary group for user (%s) is (%s)", username, primaryGroup)
 
-	return
+	// Set ownership of the SSH directory
+	ownership := fmt.Sprintf("%s:%s", username, primaryGroup)
+	if err := shell.ExecuteLive(squashErrors, "chown", "-R", ownership, userSSHKeyDir); err != nil {
+		return err
+	}
+
+	// Set permissions of the SSH directory
+	if err := shell.ExecuteLive(squashErrors, "chmod", "-R", sshDirectoryPermission, userSSHKeyDir); err != nil {
```

Comment:
Changing the error flow-style is making the diff larger than it needs to be.

Also, the Mariner toolkit generally doesn't place operations within `if` statements. So, it would probably be good to remain consistent with the rest of the codebase.